src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,73 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import { Loader } from '../Loader';
+import { useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const dispatch = useDispatch();
+  const selectedTodo = useAppSelector((state) => state.currentTodo);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [tempUser, setTempUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (selectedTodo) {
+      getUser(selectedTodo.userId)
+        .then(setTempUser)
+        .catch(() => setError(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, []);
```

Comment:
probably we want to put selectedTodo in the dependency array, what do you think?